odoo/orm/environments.py

```diff
@@ -461,14 +476,15 @@ def get(key, get_context=self.context.get):
             self._cache_key[field] = result
             return result
 
-    def flush_query(self, query: SQL):
+    def flush_query(self, query: SQL) -> None:
         """ Flush all the fields in the metadata of ``query``. """
         fields_to_flush = tuple(query.to_flush)
         if not fields_to_flush:
             return
 
-        fnames_to_flush = defaultdict(OrderedSet)
+        fnames_to_flush: defaultdict[str, OrderedSet[str]] = defaultdict(OrderedSet)
```

Comment:
This typing info has no added value to me.

Reply:
I'm keeping this one as it allows for clear type checking inside that function. Most of the time static checking needs annotations for generic types (all collections) to validate whether calls make sense.